lib/vault.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const
+  crypto = require('crypto'),
+  fs = require('fs'),
+  _ = require('lodash');
+
+class Vault {
+  /**
+   * Prepare crypto primitives.
+   * Use the key passed in parameter
+   * environment variable.
+   *
+   * @param {string|null} vaultKey - key used to decrypt the secrets
+   */
+  constructor (vaultKey = null, secretsFile = null, encryptedSecretsFile = null) {
+    this.secrets = {};
+    this.secretsFile = secretsFile;
+    this.encryptedSecretsFile = encryptedSecretsFile;
+
+    if (_.isString(vaultKey) && vaultKey.length > 0) {
+      this.vaultKey = vaultKey;
+    } else {
+      return;
+    }
+
+    this.vaultKeyHash = crypto.createHash('md5')
+      .update(this.vaultKey, 'utf-8')
+      .digest('hex')
+      .toUpperCase();
+  }
+
+  /**
+   * Decrypt secrets and store it
+   *
+   * Rejects with an error if:
+   *  - a key is found but no file
+   *  - a file is found but no key
+   */
+  decrypt () {
+    const fileExists = fs.existsSync(this.encryptedSecretsFile);
+
+    if (!fileExists) {
+      const e = new Error('No secrets file can be found.');
+
+      e.name = 'PreconditionError';
+      throw e;
+    }
+
+    const encryptedSecrets = this._readJsonFile(this.encryptedSecretsFile);
+    this.secrets = this._decryptObject(encryptedSecrets);
+  }
+
+  /**
+   * Encrypt the secrets and store it in a file
+   *
+   * Rejects with an error if:
+   *  - a key is found but no file
+   *  - a file is found but no key
+   *
+   * @param {string|null} secretsFile - file containing the encrypted secrets
+   */
+  encrypt (outputFile = null, replaceFileIfExist = false) {
+    if (!outputFile && !this.encryptedSecretsFile) {
+      const error = new Error('No outputFile or encryptedSecretsFile specified.');
+
+      error.name = 'PreconditionError';
```

Comment:
Why do we want to (incompletely) mimic a KuzzleError object here? Can't we just throw a normal error, and let Kuzzle decide what kind of error it should forward to users? As for the CLI, I don't think it needs KuzzleError instances.

Reply:
I do this so Kuzzle can identify what kind of error this is and avoid doing a
```js
if (error.message === 'No outputFile or encryptedSecretsFile specified.') {
```